### PR TITLE
feat: add stripe-cli fixtures for creating products and prices

### LINKED ISSUE #26: Add stripe-cli fixtures for creating products & prices
# Feature request

## Is your feature request related to a problem? Please describe.

When you deploy the starter it doesn't have any products and prices because the Stripe account doesn't have any.

## Describe the solution you'd like

Create a fixture json file to bootstrap some products & prices: https://stripe.com/docs/cli/fixtures

### PR DESCRIPTION
**Pull request by needcaffeine**
Created a fixtures file and added instructions to the README.

This closes #26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,15 +52,21 @@ For example, you can create business models with different pricing tiers, e.g.:
   - Price 1: 20 USD per month
   - Price 2: 200 USD per year
 
+#### Doing this automatically with fixtures
```
Comment (thorwebdev) [inline README.md:55]:
```suggestion
#### Generate test data with the Stripe CLI
```

File: README.md [hunk 2]
```diff
@@ -52,15 +52,21 @@ For example, you can create business models with different pricing tiers, e.g.:
...
+
+The Stripe cli `fixtures` command lets you run a json file as a series of API requests. We have provided you with a [fixtures file](fixtures/stripe-fixtures.json) that you can use to automatically bootstrap some products and prices in your Stripe account. This can be run with `stripe fixtures fixtures/stripe-fixtures.json`.
```
Comment (thorwebdev) [inline README.md:57]:
```suggestion
The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in a JSON file. Top help speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. Simply run `stripe fixtures fixtures/stripe-fixtures.json`.
```

File: fixtures/stripe-fixtures.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                "name": "Freelancer",
+                "description": "Freelancer product description",
+                "metadata": {
+                    "id": "prod_freelancer"
+                }
+            }
+        },
+        {
+            "name": "plan_hobby",
+            "path": "/v1/plans",
```
Comment (thorwebdev) [inline fixtures/stripe-fixtures.json:32]: Plans are outdated, can you change to prices instead, please? https://stripe.com/docs/api/prices

**Review by thorwebdev — CHANGES_REQUESTED** [review]
Thanks for this! Plans are now outdated, would you mind updating to Prices instead, please?

**Comment by vic-eb** [comment]
@thorwebdev This took a few months, but I've now updated fixtures to use the Prices API.

**Comment by needcaffeine** [comment]
@thorwebdev This took a few months, but I've now updated it to use the Prices API.

**Comment by needcaffeine** [comment]
@thorwebdev @leerob Can I get some eyes on this please? I'm afraid of being asked for changes and then forgetting about this PR for 3 more months 😂

File: README.md [hunk 4]
```diff
@@ -54,15 +54,21 @@ For example, you can create business models with different pricing tiers, e.g.:
   - Price 1: 20 USD per month
   - Price 2: 200 USD per year
 
+#### Generate test data with the Stripe CLI
+
+The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in a JSON file. Top help speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. Simply run `stripe fixtures fixtures/stripe-fixtures.json`.
```
Comment (leerob) [inline README.md:59]:
```suggestion
The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in a JSON file. To speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. Simply run `stripe fixtures fixtures/stripe-fixtures.json`.
```

File: README.md [hunk 5]
```diff
@@ -54,15 +54,21 @@ For example, you can create business models with different pricing tiers, e.g.:
...
+
+**Important:** Be sure to start the webhook forwarding (see below) so that the products created by the fixtures command above are imported into your database.
```
Comment (leerob) [inline README.md:61]: Shouldn't this come _after_ the webhook section then?

Comment (thorwebdev) [reply README.md:61]: Yes, this should be moved underneath the webhooks setup section

File: fixtures/stripe-fixtures.json [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                "currency": "usd",
+                "billing_scheme": "per_unit",
+                "unit_amount": 20000,
+                "recurring": {
+                    "interval": "month",
+                    "interval_count": 1
+                },
+                "product": "${prod_freelancer:id}"
+            }
+        }
```
Comment (thorwebdev) [inline fixtures/stripe-fixtures.json:89]: Sorry, but all plans should be removed and replaced with prices.

**Review by thorwebdev — CHANGES_REQUESTED** [review]

**Comment by dstroot** [comment]
```json
{
  "_meta": {
    "template_version": 0
  },
  "fixtures": [
    {
      "name": "prod_developer",
      "path": "/v1/products",
      "method": "post",
      "params": {
        "name": "Developer",
        "type": "service",
        "description": "Developer plans are great to learn our product.",
        "metadata": {
          "id": "prod_KJ8Tk6Jlqvf1Sf"
        }
      }
    },
    {
      "name": "price_developer_month",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_developer:id}",
        "unit_amount_decimal": "1000",
        "recurring": {
          "interval": "month",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    },
    {
      "name": "price_developer_annual",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_developer:id}",
        "unit_amount_decimal": "12000",
        "recurring": {
          "interval": "year",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    },
    {
      "name": "prod_startup",
      "path": "/v1/products",
      "method": "post",
      "params": {
        "name": "Startup",
        "type": "service",
        "description": "All the basics for starting a new business!",
        "metadata": {
          "id": "prod_KJ8Tk6Jlqvdswa"
        }
      }
    },
    {
      "name": "price_startup_month",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_startup:id}",
        "unit_amount_decimal": "2000",
        "recurring": {
          "interval": "month",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    },
    {
      "name": "price_startup_annual",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_startup:id}",
        "unit_amount_decimal": "24000",
        "recurring": {
          "interval": "year",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    },
    {
      "name": "prod_enterprise",
      "path": "/v1/products",
      "method": "post",
      "params": {
        "name": "Enterprise",
        "type": "service",
        "description": "24/7 support, and an account manager!",
        "metadata": {
          "id": "prod_KJ8Tk6Jlqvgfts"
        }
      }
    },
    {
      "name": "price_enterprise_month",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_enterprise:id}",
        "unit_amount_decimal": "4000",
        "recurring": {
          "interval": "month",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    },
    {
      "name": "price_enterprise_annual",
      "path": "/v1/prices",
      "method": "post",
      "params": {
        "product": "${prod_enterprise:id}",
        "unit_amount_decimal": "48000",
        "recurring": {
          "interval": "year",
          "interval_count": "1",
          "trial_period_days": "0",
          "usage_type": "licensed"
        },
        "currency": "usd",
        "billing_scheme": "per_unit",
        "expand": ["tiers"]
      }
    }
  ]
}
```

This is the fixture with prices - cheers!

**Comment by leerob** [comment]
Hey! Wanted to follow up here 😄 Thank you again for this.

File: README.md (hunk 5 above)
Comment (needcaffeine) [reply README.md:61]: Do you mean this entire section?

**Comment by needcaffeine** [comment]
Hey @leerob and @thorwebdev - I've fixed the conflict and pushed this back up. There is one lingering question about the "Be sure to start the webhook forwarding" section, but otherwise I believe this is done.

**Comment by leerob** [comment]
Did you see the comment above about using the Prices API instead? https://github.com/vercel/nextjs-subscription-payments/pull/58#issuecomment-929606645

**Comment by needcaffeine** [comment]
@leerob Yes, and this PR already includes a commit to change from plans to prices, just not as the original first commit.

**Comment by leerob** [comment]
"All plans should be removed and replaced with prices." - could you update with the fixtures linked above?

**Comment by needcaffeine** [comment]
oh hell I just realized there are lingering plans in that fixtures file /facepalm

**Comment by needcaffeine** [comment]
Fixtures fixed and commits squashed. Ran the fixtures:

```
❯ stripe fixtures fixtures/stripe-fixtures.json
⣻ Checking for new versions... A newer version of the Stripe CLI is available, please update to: v1.7.8
Setting up fixture for: prod_hobby
Running fixture for: prod_hobby
Setting up fixture for: price_hobby_month
Running fixture for: price_hobby_month
Setting up fixture for: price_hobby_year
Running fixture for: price_hobby_year
Setting up fixture for: prod_freelancer
Running fixture for: prod_freelancer
Setting up fixture for: price_freelancer_month
Running fixture for: price_freelancer_month
Setting up fixture for: price_freelancer_year
Running fixture for: price_freelancer_year
```

The products and prices got created:
image
image
image

**Review by leerob — APPROVED** [review]

**Pull request merged**